Bug fix tank volume nomenclature

ssankalp26 opened the pull request:
This fixes issues with nomenclature in the tank volumes.

1. Created a Volume Properties Class
2. Changed how aspect ratio is calculated

ssankalp26 committed:
initial commit to account for changes

ssankalp26 committed:
bugs

ssankalp26 committed:
Refactor fuel tank volume and mass calculations

Reworked logic for computing fuel tank volumes and masses in both fuselage and wing integral tank functions. Now accumulates total fuel mass and volume across segments, updates center of gravity calculations, and validates specified fuel mass against tank capacity. Simplifies and unifies handling of fuel properties for segmented and non-segmented cases.

Co-Authored-By: Dr. Matthew Clarke <25692010+mclarke2@users.noreply.github.com>

ssankalp26 committed:
Merge branch 'develop' into bug_fix-tank_volume_nomenclature

ssankalp26 committed:
Refactor volume properties and update fuel tank logic

ssankalp26 committed:
Refactor fuel tank volume calculations and logic

Added geometry_type attribute to Non_Integral_Tank and refactored volume computation logic to handle prismatic and cylindrical tanks separately. Improved mass and volume consistency checks, unified center of gravity assignments, and updated compute functions to return and aggregate per-segment volumes for wing tanks. These changes enhance clarity, flexibility, and correctness in fuel tank modeling.

ssankalp26 committed:
Refactor non-integral tank volume computation methods

Renamed and updated functions for prismatic tank volume calculation, improved naming consistency, and replaced deprecated attributes. Updated logic for setting fuel tank volume and mass properties, and improved error handling and output variable names for clarity. Fixed minor typos and improved comments for maintainability.

ssankalp26 committed:
changed how the length is stored to be inline with what the industry uses

ssankalp26 committed:
Add geometry type check for LH2 tank and update fuel volume call

Added a check for 'geometry_type' == 'cylinrical' in the Liquid_Hydrogen_Tank class to control tank volume and performance computations. Also removed the 'update_max_fuel' argument from the compute_fuel_volume call in geometry.py for consistency with the updated function signature.

ssankalp26 committed:
Fixed no integral fuel calculations

ssankalp26 committed:
Refactored bwb aft tank and non integral tank volume computations

ssankalp26 committed:
Fixed Regressions

ssankalp26 committed:
tutorials fix

ssankalp26 committed:
Update VERSION

ssankalp26 committed:
Merge branch 'develop' into bug_fix-tank_volume_nomenclature

ssankalp26 committed:
Add compute_fuel_tank_properties method for fuel tanks

Introduced compute_fuel_tank_properties.py to encapsulate fuel tank property calculations and integrated it into the Fuel_Tank component. Updated the __init__.py to expose the new method and incremented the version to 1.2.13.

ssankalp26 committed:
Fix mass assignment when aircraft mass is zero

Changed the check from None to zero for aircraft.mass_properties.mass to ensure correct mass assignment when the value is zero. This improves robustness in handling aircraft mass properties.

ssankalp26 committed:
Regression Fixes and patch test

ssankalp26 committed:
Temporary fix for rotor res files needing volume properties

ssankalp26 committed:
temporary fix for fuel tank volume bug with plot 3d

mclarke2 commented on RCAIDE/Library/Components/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank.py:
```diff
@@ -100,12 +102,14 @@ def __defaults__(self):
         self.aft_tank_end_segment_tag    = None 
         self.wing_root_tag               = None 
-        self.radial_offset               = 0.0
+        self.radial_offset               = None
```
what is radial offset? please define it

ssankalp26 replied:
Done

mclarke2 commented on VnV/Vehicles/Tiltrotor_EVTOL.py:
```diff
@@ -540,7 +540,8 @@ def vehicle_setup(redesign_rotors=True) :
         loaded_propulsor = load_propulsor(os.path.join(test_dir, 'proprotor_propulsor.res'))  
         for key,item in propulsor.rotor.items():
-            propulsor.rotor[key] = loaded_propulsor.rotor[key] 
+            if key != "volume_properties":
```
what makes volume properties different from the rest of the other keys?

mclarke2 commented on VnV/Vehicles/Tiltrotor_EVTOL.py:
```diff
@@ -553,7 +554,8 @@ def vehicle_setup(redesign_rotors=True) :
        
         for key,item in propulsor.motor.items():
-            propulsor.motor[key] = loaded_propulsor.motor[key] 
+            if key != "volume_properties":
```
See comment above

mclarke2 commented on VnV/Vehicles/Tiltwing_EVTOL.py:
```diff
@@ -318,7 +318,8 @@ def vehicle_setup(new_regression=True):
         for key,item in prop_rotor_propulsor.rotor.items():
-            prop_rotor_propulsor.rotor[key] = loaded_propulsor.rotor[key] 
+            if key != "volume_properties":
+                prop_rotor_propulsor.rotor[key] = loaded_propulsor.rotor[key] 
```
This is buggy and the start of something we don't want to go down. I suggest you remove these lines or find a clearer fix

mclarke2 commented on VnV/Verification/analysis_aerodynamics/avl_files_surrogate/batch_0006.run:
```diff
@@ -124,7 +124,7 @@
  X_cg      =   2.239696797     m
  Y_cg      =   0     m
  Z_cg      =   -0.131189711     m
- mass      =   0.0     kg
```
The AVL files cannot be done, this is going to create a bug when running AVL

mclarke2 commented on VnV/Verification/network_turbojet/turbojet_network_errors.res:
```diff
@@ -1 +1 @@
-{"CL": 1.577911775087201e-15}
```
Why did this change?

mclarke2 reviewed (commented):
Just a few major things to address, good job

mclarke2 commented on RCAIDE/Library/Methods/Geometry/Planform/compute_fuel_volume.py:
```diff
@@ -81,19 +63,30 @@ def compute_fuel_volume(vehicle, update_max_fuel=True):
-                    pass
+                    net_volume = getattr(fuel_tank.fuel.volume_properties, "net_volume", None)
+                    if net_volume is None:
+                        raise ValueError("Net volume is not defined for this fuel tank")
```
This is a lot of verbiage... is this necessary? also brings coverage down. Really, you dont need any of this at all

ssankalp26 committed:
Add volume properties to Vehicle and Component classes

Introduces Volume_Properties to the Component class and Vehicle_Volume_Properties to the Vehicle class. Also initializes the fuel attribute in Vehicle_Volume_Properties and ensures volume properties are appended to the vehicle components.

mclarke2 committed:
updates

mclarke2 committed:
updated to blade optimization

mclarke2 committed:
Update plot_3d_vehicle.py

ssankalp26 committed:
Fixes except plot and rotor

ssankalp26 committed:
Refactor geometry plotting and update test truth values

Refactored plot_3d_vehicle and related functions to use 'geometry' instead of 'vehicle' for clarity and consistency. Added plot_tank_geometry flag to control fuel tank plotting. Updated truth values in turbofan_network_test.py and rotor_performance_test.py to reflect new reference results.

ssankalp26 committed:
Update turbojet_network_test.py

ssankalp26 committed:
Set rotor_test new_regression to False in main

Changed the rotor_test function call in main() to use new_regression=False instead of True. This likely switches the test mode from creating a new regression baseline to using an existing one.

ssankalp26 committed:
updates

ssankalp26 committed:
Fix tank origin calculation and update plot_3d_vehicle args

Corrects the z-coordinate calculation for the fuel tank origin in compute_bwb_aft_tank_volume by subtracting half the outer diameter. Also adds the plot_tank_geometry argument to the plot_3d_vehicle function call.

ssankalp26 committed:
Update fuel_tank_volume_test.py

ssankalp26 committed:
improving coverage

ssankalp26 committed:
Update test_automatic_regression.py

ssankalp26 committed:
Update test_automatic_regression.py

mclarke2 commented on VnV/Vehicles/Tiltwing_EVTOL.py:
```diff
@@ -317,7 +317,7 @@ def vehicle_setup(new_regression=True):
         regression_prop_rotor_propulsor = deepcopy(prop_rotor_propulsor)        
         design_electric_rotor(regression_prop_rotor_propulsor, iterations=2)
```
add print_iterations here

ssankalp26 committed:
Enable iteration printing in design_electric_rotor calls

Added the print_iterations=True argument to design_electric_rotor function calls in vehicle_setup to provide iteration output during rotor design and regression processes.

mclarke2 commented on VnV/Verification/mission_segments/active_transition_test.py:
```diff
@@ -28,7 +28,7 @@
 def main(): 
     # make true only when resizing aircraft. should be left false for regression
-    update_regression_values = False
+    update_regression_values = True
```
Make this False

mclarke2 replied:
I resolved this

mclarke2 committed:
Update active_transition_test.py

ssankalp26 committed:
bug fixes

ssankalp26 committed:
Merge branch 'bug_fix-tank_volume_nomenclature' of https://github.com/leadsgroup/RCAIDE_LEADS into bug_fix-tank_volume_nomenclature

MGuidotti committed:
Delete Offtake_Shaft.py

ssankalp26 committed:
Turboelectricgenerator

MGuidotti committed:
Refactor turboelectric generator data reuse logic

Updated the reuse_stored_data and reuse_stored_turboelectric_generator_data functions to use network and stored_converter_tag instead of fuel_line for retrieving stored converter data. Adjusted attribute access to use nested turboshaft components, improving clarity and correctness in data copying for identical propulsors.

ssankalp26 committed:
Working Turboelectric Generator Reuse Data Segment

ARmolloy committed:
Diameter to radius update

ssankalp26 committed:
Add Cycom 5320 material and update LH2 tank logic

Introduces the Cycom_5320 material class and registers it in the materials module. Updates the Liquid_Hydrogen_Tank default inlet temperature from 15 to 20. Adjusts thermal performance computation to double insulation mass for symmetric tanks. Adds Cycom_5320 to the coverage test. Updates a test result in turbojet_network_errors.res.

ssankalp26 committed:
Merge branch 'bug_fix-tank_volume_nomenclature' of https://github.com/leadsgroup/RCAIDE_LEADS into bug_fix-tank_volume_nomenclature
